### PR TITLE
Minor improvements and updates to readthedocs.io and mkdocs configs

### LINKED ISSUE #9333: Update readthedocs to version 2
## Description

Readthedocs configuration is deprecated and it's support will be removed in septenber 2023.
https://blog.readthedocs.com/migrate-configuration-v2/

https://docs.readthedocs.io/en/stable/config-file/v1.html

We need to update the configuration to version 2. 

**What kind of improvement you want to add?** (check one with "x", remove the others)
- [ ] Minor changes to existing features
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] CI related changes
- [ ] Other... Please describe:

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
This PR tries to fix readthedocs build. It can be used to work on the solution.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [x] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

Fix #9333

**What is the new behavior?**

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Comment by offtherailz** [comment]
I fixed the files to make the build work, but I'm afraid they did some issues on readthedocs.io too. 
I fixed the files as suggested by the build failure, then I had a memory consumption error. I fixed all the warnings, expecially ones related to PDF, because I though it was the PDF generation to generate the memory issue. 

In fact now the build run but it takes a lot (30 minutes...) so I'm wandering if there is something else on the readthedocs side...

**Comment by offtherailz** [comment]
The issue seems to be solved. 
This PR contains some updates/fixes to documentation warnings. We can take this opportunity to apply them.

File: .readthedocs.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details
+
+# Required
+version: 2
+
+# Set the version of Python and other tools you might need
+build:
+  os: ubuntu-22.04
+  tools:
+    python: "3.11"
```
Comment (offtherailz) [inline .readthedocs.yaml:11]:
```suggestion
    python: "3.10.6"
```

File: .readthedocs.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details
+
+# Required
+version: 2
+
+# Set the version of Python and other tools you might need
+build:
+  os: ubuntu-22.04
+  tools:
+    python: "3.10.6"
```
Comment (offtherailz) [inline .readthedocs.yaml:11]:
```suggestion
    python: "3.10"
```

File: .readthedocs.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details
+
+# Required
+version: 2
+
+# Set the version of Python and other tools you might need
+build:
+  os: ubuntu-22.04
+#  tools:
+#    python: "3.10"
```
Comment (offtherailz) [inline .readthedocs.yaml:11]:
```suggestion
  tools:
    python: "3.7"
```

File: .readthedocs.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details
+
+# Required
+version: 2
+
+# Set the version of Python and other tools you might need
+build:
+  os: ubuntu-22.04
+  tools:
+    python: "3.7"
```
Comment (offtherailz) [inline .readthedocs.yaml:11]:
```suggestion
# Adding the following, ask proposed by readthedocs.io documentation
# makes the build fail after a while for eccessive memory consumption
# See https://github.com/readthedocs/readthedocs.org/issues/10585
# build:
#  os: ubuntu-22.04
#  tools:
#    python: "3.7"
```

**Comment by offtherailz** [comment]
The readthedocs.io has been updated to version 2

Anyway I encountered this issue: 
https://github.com/readthedocs/readthedocs.org/issues/10585

File: build/tests.webpack.js [hunk 5]
```diff
@@ -1,3 +1,3 @@
-var context = require.context('../web', true, /(-test\.jsx?)|(-test-chrome\.jsx?)$/);
+var context = require.context('../web/', true, /(-test\.jsx?)|(-test-chrome\.jsx?)$/);
```
Comment (offtherailz) [inline build/tests.webpack.js:1]:
```suggestion
var context = require.context('../web', true, /(-test\.jsx?)|(-test-chrome\.jsx?)$/);
```

**Comment by offtherailz** [comment]
here the last attempt to build using standard readthedocs build in yaml

File: .readthedocs.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+version: 2
+
+# Set the version of Python and other tools you might need
+# Adding the following, ask proposed by readthedocs.io documentation
+# makes the build fail after a while for eccessive memory consumption
+# See https://github.com/readthedocs/readthedocs.org/issues/10585
+build:
+ os: ubuntu-22.04
+ tools:
+   python: "3.7"
```
Comment (offtherailz) [inline .readthedocs.yaml:14]:
```suggestion
# build:
#  os: ubuntu-22.04
#  tools:
#    python: "3.7"
```

File: docs/requirements.txt [hunk 7]
```diff
@@ -3,3 +3,4 @@ jinja2==3.1.2
 Markdown==3.4.1
 git+https://github.com/geosolutions-it/mkdocs-pdf-export-plugin.git
 WeasyPrint==52.5
+mkdocs==1.5.2
```
Comment (offtherailz) [inline docs/requirements.txt:6]:
```suggestion
```

**Review by MV88 — APPROVED** [review]
LGTM

**Pull request merged**